### PR TITLE
API client factory (TKX-2275)

### PR DESCRIPTION
**Pull request by joaofbvieira**

### CODE REVIEW

File: src/Trakx.Fireblocks.ApiClient/FireblocksApiClients.Partials.tt
Comment (joaofbvieira) [inline src/Trakx.Fireblocks.ApiClient/FireblocksApiClients.Partials.tt:1]: Some clients were obsolete, so they were removed.

File: src/Trakx.Fireblocks.ApiClient/FireblocksApiClients.Partials.cs
Comment (joaofbvieira) [inline src/Trakx.Fireblocks.ApiClient/FireblocksApiClients.Partials.cs:1]: Just property renaming.

File: src/Trakx.Fireblocks.ApiClient/AuthorisedClient.cs
Comment (joaofbvieira) [inline src/Trakx.Fireblocks.ApiClient/AuthorisedClient.cs:1]: Just property renaming.

File: src/Trakx.Fireblocks.ApiClient/FireblocksApiClientsFactory.cs
Comment (joaofbvieira) [inline src/Trakx.Fireblocks.ApiClient/FireblocksApiClientsFactory.cs:1]: Implementation inspired from `ExchangeApiClientsFactory`, reference [here](https://github.com/trakx/exchange-api-client/blob/dev/src/Trakx.Exchange.ApiClient/Factories/ExchangeApiClientsFactory.cs).

File: tests/Trakx.Fireblocks.ApiClient.Tests/Unit/AddFireblocksClientExtensionTests.cs
Comment (joaofbvieira) [inline tests/Trakx.Fireblocks.ApiClient.Tests/Unit/AddFireblocksClientExtensionTests.cs:1]: DI registration unit tests reviewed.

**Review by monsieurleberre — APPROVED** [review]
Thanks!

**Pull request merged**